esrally/chart_generator.py

```diff
@@ -1395,28 +1596,66 @@ def load_race_configs(cfg):
     return race_configs
 
 
+def gen_charts_per_track_configs(race_configs, chart_type, env, flavor=None, logger=None):
+    charts = generate_index_ops(chart_type, race_configs, env, logger) + \
+             generate_io(chart_type, race_configs, env) + \
+             generate_gc(chart_type, race_configs, env) + \
+             generate_segment_memory(chart_type, race_configs, env) + \
+             generate_queries(chart_type, race_configs, env)
+
+    dashboard = generate_dashboard(chart_type, env, race_configs[0].track, charts, flavor)
+
+    return charts, dashboard
+
+
+def gen_charts_per_track(race_configs, chart_type, env, flavor=None, logger=None):
+    structures = []
+    for race_configs_per_track in race_configs:
+        charts, dashboard = gen_charts_per_track_configs(race_configs_per_track, chart_type, env, flavor, logger)
+        structures.extend(charts)
+        structures.append(dashboard)
+
+    return structures
+
+
+def gen_charts_from_track_combinations(race_configs, chart_type, env, logger):
+    structures = []
+    for flavor, race_configs_per_flavor in race_configs.items():
+        for race_configs_per_track in race_configs_per_flavor:
+            logger.debug("Generating charts for race_configs with name:[{}]/flavor:[{}]".format(race_configs_per_track[0].name, flavor))
```

Comment:
Our convention is to use plain logger strings for log statements instead of the format API.